### PR TITLE
RT-2.9: IS-IS metric style wide enabled

### PR DESCRIPTION
**Pull request by divyarema**
initial commit for RT2.9
It resolves https://github.com/openconfig/featureprofiles/issues/1618

### CODE REVIEW

**Review by greg-dennis — CHANGES_REQUESTED** [review]

**Review by sachendras — APPROVED** [review]

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+			}
+		})
+		t.Run("Traffic checks", func(t *testing.T) {
+			ate.Traffic().Start(t, flows...)
+			time.Sleep(time.Second * 15)
+			ate.Traffic().Stop(t)
+
+			for _, flow := range flows {
+				t.Log("Checking flow telemetry...")
+				telem := gnmi.OC()
```
[the remark this answers is not in the document]
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+	netGrp.ISIS().WithActive(true).WithIPReachabilityMetric(ateV6Metric)
+
+	t.Log("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return topo
+}
+
+// createFlow returns v4 and v6 flow from atePort2 to atePort1
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, ateTopo *ondatra.ATETopology) []*ondatra.Flow {
```
[the remark this answers is not in the document]
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# RT-2.9: IS-IS  metric style wide enabled
+
+## Summary
+
+*  Base IS-IS functionality and adjacency establishment.
+*  Verifies route metric with wide metric enabled on DUT.
+
+## Procedure
+
+*   TestISISWideMetricEnabled
```
[the remark this answers is not in the document]
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/README.md]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+		fptest.SetPortSpeed(t, dut.Port(t, "port2"))
+	}
+}
+
+// configureISIS configures isis on DUT.
+func configureISIS(t *testing.T, dut *ondatra.DUTDevice, intfName string, dutAreaAddress, dutSysID string) {
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go:112]: remove the dutAreaAddress and dutSysID parameters and just access the globals instead

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+	ateTopo := configureATE(t, ate)
+	flows := createFlows(t, ate, ateTopo)
+
+	if deviations.ExplicitInterfaceInDefaultVRF(dut) {
+		intfName = intfName + ".0"
+	}
+	statePath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_ISIS, isisInstance).Isis()
+
+	t.Run("ISIS telemetry", func(t *testing.T) {
+
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go:273]: delete blank line

**Review by greg-dennis — DISMISSED** [review]
Two small requests

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go (hunk 5 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go]: corrected

File: feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go (hunk 4 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_metric_style_wide_enabled_test/isis_metric_style_wide_enabled_test.go]: addressed

**Review by arulkumarsekar — APPROVED** [review]

**Pull request merged**